### PR TITLE
[Tcs] Add app bind/unbind & Set patterns in app update

### PR DESCRIPTION
**Pull request by ninpan-ms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/tanzu.py [hunk 1]
```diff
@@ -205,3 +226,30 @@ def _assert_deployment_exist_and_retrieve_name(cmd, client, resource_group, serv
...
+                TANZU_CONFIGURATION_SERVICE_PROPERTY_PATTERN: patterns
+            }
+        )
+    }
+
+
+def _tcs_bind_or_unbind_app(cmd, client, resource_group, service, app_name, enabled):
+    try:
+        app = _app_get(cmd, client, resource_group, service, app_name)
+        previous = app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME].enabled
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/tanzu.py]: check null for addon_config and whether contains key

File: src/spring-cloud/azext_spring_cloud/tanzu.py [hunk 2]
```diff
@@ -205,3 +226,30 @@ def _assert_deployment_exist_and_retrieve_name(cmd, client, resource_group, serv
...
+        app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME] = models.AddonProfile(
+            enabled=enabled
+        )
+        if (enabled != previous):
+            return client.apps.create_or_update(resource_group, service, app_name, app.properties)
+        elif (enabled):
+            logger.warning('app {} has been binded'.format(app_name))
+        else:
+            logger.warning('app {} has been unbinded'.format(app_name))
+    except CloudError:
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/tanzu.py]: `_app_get` will throw CLIError App not found. no need to try catch here.

File: src/spring-cloud/azext_spring_cloud/tanzu.py [hunk 3]
```diff
@@ -205,3 +226,30 @@ def _assert_deployment_exist_and_retrieve_name(cmd, client, resource_group, serv
...
+def _tcs_bind_or_unbind_app(cmd, client, resource_group, service, app_name, enabled):
+    try:
+        app = _app_get(cmd, client, resource_group, service, app_name)
+        previous = app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME].enabled
+        app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME] = models.AddonProfile(
+            enabled=enabled
+        )
+        if (enabled != previous):
+            return client.apps.create_or_update(resource_group, service, app_name, app.properties)
+        elif (enabled):
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/tanzu.py]: nit: `logger.warning('app {} has been {}binded.'.format(app_name, '' if enabled else 'un'))`

**Review by yuwzho — COMMENTED** [review]
Please also add helper message in `src/spring-cloud/azext_spring_cloud/_help.py`

File: src/spring-cloud/azext_spring_cloud/tanzu.py (hunk 2 above)
Comment (ninpan-ms) [reply src/spring-cloud/azext_spring_cloud/tanzu.py]: Yes, but the error will be "App not found", without app name.

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 4]
```diff
@@ -19,7 +19,6 @@
 
 logger = get_logger(__name__)
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_validators.py]: nit: keep two white lines before the method to make lint pass.

**Review by yuwzho — APPROVED** [review]
Don't forget the help.py file for the two new commands :)

File: src/spring-cloud/azext_spring_cloud/_help.py [hunk 5]
```diff
@@ -86,6 +86,21 @@
       text: az spring-cloud tanzu app deploy -g myresourcegroup -s myservice -n myapp --artifact-path <my-Jar>
 """
 
+helps['spring-cloud tanzu application-configuration-service'] = """
+    type: group
+    short-summary: Commands to manage Azure Spring Cloud Tanzu application-configuration-service.
+"""
+
+helps['spring-cloud tanzu application-configuration-service bind'] = """
+    type: group
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_help.py]: command

File: src/spring-cloud/azext_spring_cloud/_help.py [hunk 6]
```diff
@@ -86,6 +86,21 @@
...
+    short-summary: Commands to manage Azure Spring Cloud Tanzu application-configuration-service.
+"""
+
+helps['spring-cloud tanzu application-configuration-service bind'] = """
+    type: group
+    short-summary: Bind app to Tanzu Application Configuration Service.
+"""
+
+helps['spring-cloud tanzu application-configuration-service unbind'] = """
+    type: group
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_help.py]: command

**Pull request merged**